cogs/fishing.py

```diff
@@ -227,21 +222,25 @@ async def fish(self, ctx: commands.Context):
 
         # See which fish they caught
         new_fish = random.choice(list(self.bot.fish[rarity].values())).copy()
-        if special == "inverted":
-            new_fish = utils.make_inverted(new_fish)
-        elif special == "golden":
-            new_fish = utils.make_golden(new_fish)
+        
+        special_functions = {
+            "inverted": utils.make_inverted(new_fish),
+            "golden": utils.make_golden(new_fish)
+        }
+        
+        if special in special_functions.keys():
+            new_fish = special_functions[special]    
```

Comment:
This won't work - it'll run `make_inverted` and `make_golden` every time. If you don't want that then you should wrap it in a function and call it as necessary, but I don't think this improves readability anyway.

Reply:
Why will it run it every time? Wouldn't it just run it if the special word is in the dict and then run just one of the functions? It's pretty similar to how it was before but this time with a dictionary

Reply:
The dictionary gives a value, not a "state". `inverted` will have the value returned by `utils.make_inverted`, for example, and to get the value the function needs to be run.

![image](https://user-images.githubusercontent.com/5342536/120854649-5bfbb480-c575-11eb-95c9-5175b7a27fec.png)

Reply:
new_fish should be the value returned by make_inverted or make_golden.

the make_... commands return a dict where the name of the fish is either "inverted_fishname" or golden.

Either this does that or I'm very confused lol

Reply:
Ah. Yes you're normally right. These functions *modify* the input dictionary though, not just return a new one.